src/stream/src/executor/global_simple_agg.rs

```diff
@@ -96,19 +101,23 @@ impl<S: StateStore> GlobalSimpleAggExecutor<S> {
         ctx: ActorContextRef,
         input: Box<dyn Executor>,
         agg_calls: Vec<AggCall>,
+        mut agg_state_tables: Vec<Option<AggStateTable<S>>>,
+        mut result_table: StateTable<S>,
         pk_indices: PkIndices,
         executor_id: u64,
         extreme_cache_size: usize,
-        mut state_tables: Vec<StateTable<S>>,
-        state_table_col_mappings: Vec<Vec<usize>>,
     ) -> StreamResult<Self> {
         let input_info = input.info();
         let schema = generate_agg_schema(input.as_ref(), &agg_calls, None);
 
         // TODO: enable sanity check for globle simple agg executor <https://github.com/risingwavelabs/risingwave/issues/3885>
-        for state_table in &mut state_tables {
-            state_table.disable_sanity_check();
-        }
+        agg_state_tables
+            .iter_mut()
+            .filter_map(Option::as_mut)
+            .for_each(|state_table| {
+                state_table.table.disable_sanity_check();
+            });
+        result_table.disable_sanity_check();
```

Comment:
As we have `result_table` persisted now, can we use the correct `update` to enable the sanity check? 🤔 (and even `upsert` can be removed?)

Reply:
Yep. It may be better to do this in a new PR.